yabt/buildcontext.py

```diff
@@ -419,14 +420,18 @@ def run_in_buildenv(
             except UnicodeEncodeError as e:
                 sys.stderr.write('tried writing the stdout of {},\n but it '
                                  'has a problematic character:\n {}\n'
-                                 .format(docker_run, str(e)))
+                                 'hex dump of stdout:\n{}\n'
```

Comment:
Don't you want the except after the two if's ? it would save you the code dup.

Reply:
And then print both stdout and stderr in hex dump?